.github/workflows/ci-lint.yml

```diff
@@ -19,15 +19,36 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+        with:
+          # Checkout pull request HEAD commit instead of merge commit.
+          ref: ${{ github.event.pull_request.head.sha }}
+          # Fetch all history so gitlint can check the relevant commits.
+          fetch-depth: '0'
+      - name: Set up Python 3
+        uses: actions/setup-python@v4.2.0
+        with:
+          python-version: '3.x'
```

Comment:
My preference is to isolate this in a separate job

Reply:
Let's keep it in the same job for consistency with other repos for now.